server/matchmaker/algorithm.py

```diff
@@ -151,22 +120,45 @@ def find(self) -> Dict[Search, Search]:
 
 @with_logger
 class Matchmaker(object):
-    def __init__(self, searches: List[Search]):
+    def __init__(self, searches: Iterable[Search]):
         self.searches = searches
         self.matches: Dict[Search, Search] = {}
 
     def find(self) -> List[Match]:
         self._logger.debug("Matching with stable marriage...")
-        self.matches.update(StableMarriage(self.searches).find())
+        searches = [s for s in self.searches if s is not None]
```

Comment:
How is it possible to have `None` values in that searches. Like list witho bjects and nones is bad. If this is necessary, then typing of self search in init should be iterable optional search

Reply:
TBH I have no idea how that got in there. Somehow I thought it was in the code before and I just left it alone, but looking at the diff I don't see it.